oat-sa/tao-core
# Feature/ADF-1784/Side by side authoring

### Related to: https://oat-sa.atlassian.net/browse/ADF-1784

### Summary

Add the side-by-side UI in the item authoring when translating.

<img width="1840" alt="image" src="https://github.com/user-attachments/assets/f90000dc-5fd4-4a21-bc57-9e7fdf874e73">

### Details

Add service connectors to:
- delete a translation
- update the progress

Note: the server side may need to be refactored as it is fast approach.

### How to test
- checkout the branch: `git checkout -t origin/feature/ADF-1784/side-by-side-authoring`
- also make sure to checkout the companion PRs
- make sure to have the Feature Flags activated
```
FEATURE_FLAG_TRANSLATION_ENABLED=true
FEATURE_FLAG_UNIQUE_NUMERIC_QTI_IDENTIFIER=true
FEATURE_FLAG_TRANSLATION_DEVELOPER_MODE=true
```
- make sure to have the Item Translator role assigned to your user
- have an item ready for translation and add/edit a translation: it should present the side-by-side authoring
- from a regular item, also open the authoring: it should present the usual auhoring
